### ISSUE TITLE
[BUG]

### ISSUE THREAD

**Issue by AlexandreOuellet** [longhorn/longhorn#1848]
**Describe the bug**
Longhorn seems to ignore Default Replica Count when it is set to 1 (to run in a single-node cluster).  It still schedule 3 replicas on a volume created through a volume template.

**To Reproduce**
1. Ensure that you are running on a single cluster
2. In the Longhorn UI, set the Default Replica Count to 1
3. Deploy a statefulsets to kubernetes
4. The volume will not be attached, with the error "Replica Scheduling Failure", and it seems to have 3 replica

**Expected behavior**
When I deploy through statefulsets and use VolumeClaimTemplates, I would expect Longhorn to take into account how many replica I've configured thorugh the UI

**Log**
None

**Environment:**
 - Longhorn version: 1.0.2
 - Kubernetes version: 1.19.0
 - Node OS type and version: Ubuntu 18.04

**Additional context**
If I create a volume through the UI, the volume is configured to use only a single Replica (as expected), but when the volume is created through a StatefulSets deployment, the volume is configured with 3 replicas, and is unschedulable.  I have also tried restarting all of the pods from the longhorn deployment, but I still get 3 replicas, even though I configured it to use a single one

I do not know of a workaround currently.

**Comment by AlexandreOuellet** [comment]
I believe the issue is coming from the StorageClass.  It still has paramters.numberOfReplicas set to 3, and I believe this is causing confusion with what is in the UI.

I would expect that when I modify the UI, this would also get modified, or that there is only one way to modify it (either through the storage class or the UI), but not both spot.

**Comment by yasker** [comment]
The UI's setting for replica is for UI created volume. 

```
The default number of replicas when creating the volume from Longhorn UI. For Kubernetes, update the numberOfReplicas in the StorageClass
```

Kubernetes doesn't allow modify the parameter of the storage class. So you can create another storage class with Longhorn as driver, and set replica count to 1 on that.

It's not recommended to modify the default storage class since it might cause issue during the upgrade.

**Comment by AlexandreOuellet** [comment]
That was essentially my workaround : Having a storage class with longhorn as the driver, and set the number of replica to 1 instead.  May I suggest adding clarification to the documentation for easier setup in single (or even dual) node setup?

**Comment by yasker** [comment]
Yes, we can add more clarification on running Longhorn on a smaller cluster.

**Comment by yasker** [comment]
@PhanLe1010 Do you think we might be able to update the storage class dynamically according to the setting? That's going to save lots of confusion.

**Comment by PhanLe1010** [comment]
@yasker 
Other than the issue that PVs created bu the same `longhorn` storageclass have inconsistent parameters, I don't see any other problems. Old PVs using old version of `longhorn` storageclass will continue to operate with old parameters. New PVs using new version of `longhorn` storageclass will have new parameters.

**Comment by PhanLe1010** [comment]
The implementation is also straightforward. We can create a custom controller `storageclass-controller` and listen for any setting changes. If there is a change to `longhorn` storageclass, we delete the old one and create a new one. There will be a small down time between the deletion and creation. During the down time, new PV creation will be failed

**Comment by yasker** [comment]
@PhanLe1010 We don't need a separate controller. We can do it in the setting controller. Though what if the storageclass replica count changed without the setting changed? Who is the source of the truth?

**Comment by PhanLe1010** [comment]
I see
